tests/features/camel-jackson/src/test/java/org/apache/karaf/camel/itest/CamelJacksonITest.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.karaf.camel.itest;
+
+import org.apache.camel.component.mock.MockEndpoint;
+import org.apache.karaf.camel.itests.AbstractCamelSingleFeatureResultMockBasedRouteITest;
+import org.apache.karaf.camel.itests.CamelKarafTestHint;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.ops4j.pax.exam.junit.PaxExam;
+import org.ops4j.pax.exam.spi.reactors.ExamReactorStrategy;
+import org.ops4j.pax.exam.spi.reactors.PerClass;
+
+@CamelKarafTestHint
+@RunWith(PaxExam.class)
+@ExamReactorStrategy(PerClass.class)
+public class CamelJacksonITest extends AbstractCamelSingleFeatureResultMockBasedRouteITest {
+
+    @Override
+    public void configureMock(MockEndpoint mock) {
+        mock.expectedBodiesReceived("Jack");
```

Comment:
I would more expect that you override the method `getBodyToSend` to send a JSon payload, in the producer you first unmarshal it, add assertions to check the values of `getName` and `getAge`, finally the producer marshals it, and the test checks that the result is the expected JSon payload (same as what has been sent?)

Reply:
I feel safer to have just https://github.com/apache/camel-karaf/blob/18853809d59c0da1ca5f11a0711e759c2bb6836a/tests/features/camel-jackson/src/test/java/org/apache/karaf/camel/itest/CamelJacksonITest.java#L41 

The default behaviour in the libraries might change as we upgrade them. For instance, the marshal operation could generate JSON in pretty print by default, with extra characters, which would cause our tests to fail.